fix #269672: Show selected notes on the piano control

mattmcclinch opened the pull request:
Alternative to #3488.

lasconic commented on mscore/pianotools.cpp:
```diff
@@ -167,6 +168,25 @@ void HPiano::releasePitch(int pitch)
+            key->setHighlighted(false);
+            key->setSelected(false);
+            }
+      for (Note*n : selection.uniqueNotes()) {
```
a space is missing here.

lasconic commented on mscore/pianotools.cpp:
```diff
@@ -167,6 +168,25 @@ void HPiano::releasePitch(int pitch)
+            key->setSelected(false);
+            }
+      for (Note*n : selection.uniqueNotes()) {
+            keys[n->pitch()-_firstKey]->setSelected(true);
```
Please use spaces around -

lasconic commented on mscore/pianotools.cpp:
```diff
@@ -167,6 +168,25 @@ void HPiano::releasePitch(int pitch)
+      for (Note*n : selection.uniqueNotes()) {
+            keys[n->pitch()-_firstKey]->setSelected(true);
+            for (Note* other : n->chord()->notes())
+                  keys[other->pitch()-_firstKey]->setHighlighted(true);
```
same here, spaces around -

mattmcclinch commented:
Okay, I added the missing spaces.

When I was testing this patch in 3.0, I noticed that the selected piano key did not change if I added an accidental to the selected note. I investigated, and found this in libmscore/undo.cpp:
`
//---------------------------------------------------------
//   SaveState
//---------------------------------------------------------

SaveState::SaveState(Score* s)
   : undoInputState(s), redoInputState(s->inputState())
      {
      score = s;
//      redoSelection  = score->selection();
      }

void SaveState::undo(EditData*)
      {
      redoInputState = score->inputState();
//      redoSelection  = score->selection();
      score->setInputState(undoInputState);
//      score->setSelection(undoSelection);
      }

void SaveState::redo(EditData*)
      {
      undoInputState = score->inputState();
//      undoSelection  = score->selection();
      score->setInputState(redoInputState);
//      score->setSelection(redoSelection);
      }
`

Is there any reason these lines are commented out?

lasconic commented:
No idea, @wschweer ?

lasconic commented:
I will just merge it  as it in 3.0 if possible and let's solve the problem later with @wschweer.

wschweer commented:
Its difficult to store the selection state in the undo/redo history as its hard to avoid dangling pointers to elements which are deleted during layout. So until there is a better solution i commented out the functionality.

mattmcclinch commented:
Interesting. Does version 2.x have this dangling pointer problem?

mattmcclinch commented:
Did you ever have a look at #3476? It addresses a problem similar to what you describe. It may be a step toward the solution you are looking for.
